sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -40,12 +43,25 @@ class HFTransformersForecaster(BaseForecaster):
         Path to the huggingface model to use for forecasting. Currently,
         Informer, Autoformer, and TimeSeriesTransformer are supported.
     fit_strategy : str, default="minimal"
-        Strategy to use for fitting the model. Can be "minimal" or "full"
+        Strategy to use for fitting (fine-tuning) the model. This can be one of
+        the following:
+        - "minimal": Fine-tunes only a small subset of the model parameters,
+          allowing for quick adaptation with limited computational resources.
+        - "full": Fine-tunes all model parameters, which may result in better
+          performance but requires more computational power and time.
+        - "lora": Uses the LoRA (Low-Rank Adaptation) technique to fine-tune the
+          model efficiently with low-rank updates.
+        - "loha": Similar to LoRA, but with higher-rank adaptations.
+        - "adalora": An adaptive version of LoRA that adjusts the rank
+          dynamically during fine-tuning.
     validation_split : float, default=0.2
         Fraction of the data to use for validation
     config : dict, default={}
         Configuration to use for the model. See the `transformers`
         documentation for details.
+    peft_config_dict : dict, default={}
+        Configuration dictionary specifying parameters and settings relevant to
```

Comment:
please explain possible fields, or make them available directly

Reply:
I created an example in docstring for that. Instead I think I should link here the peft configuration documentation [here](https://huggingface.co/docs/peft/en/tutorial/peft_model_config)

by "available directly" did you mean to set the default parameters to something other that {}? but the default one will not work for all peft methods as they take different params.

Reply:
> by "available directly" did you mean to set the default parameters to something other that {}? but the default one will not work for all peft methods as they take different params.

Yes, I meant as args. How bad is the variability in arguments? One option that you can take is always to take the union, and ignore arguments that are not applicable.

More quantitatively, how many config args are there in total, how many are common?

> Instead I think I should link here the peft configuration documentation

That may be a good idea. Also, are there any common config sets that we could hardcode by a string?

Reply:
First I don't think we can have params with a default value of dictionary object - flake 8 `B006 Do not use mutable data structures for argument defaults`

> More quantitatively, how many config args are there in total, how many are common?

Mostly 4-5

> are there any common config sets that we could hardcode by a string?

we can use these - https://github.com/sktime/sktime/blob/44580748a82c2c4139c6e79a05a5663b2949a9d0/sktime/forecasting/hf_transformers_forecaster.py#L234C1-L258C65

Reply:
The only important param for `peft_config` is `target_modules` which has to be a list of strings. Rest of the params can be set by default.
I suggest that we only make 2 changes here
1. add link for peft documentation
2. set the default value of `peft_config_dict` to `{"target_modules": ["q_proj", "v_proj"]}` instead of `{}`, so user can run the code without any peft_config_dict argument and can also mutate it through argument

Reply:
2. yes, default settings in estimators should *always* run, and that should be tested by `get_test_params`.

1. I would suggest to add the link, but also document the most important params.

Reply:
> First I don't think we can have params with a default value of dictionary object 

Yes, one should never use mutable defaults as they can lead to hard to diagnose errors.

The way you set a mutable default is, you set the default to `None`, and you conditionally write to a private parameter, e.g., `self._my_config`, depending on the public value of `self.my_config`.

Reply:
> how many parameters - inside the configs - are there in total?
> how many are shared for different choices of PEFT method?

15+ in total, 4-5 used commonly, 2 of which are common to all configs

Reply:
hm, I would have a light tendency towards making the common ones explicit, and pass the rest as dict.

Reply:
in the current implementation common parameter is provided as default and thus the code runs when parameters are set to default and users can provide additional params through `peft_config_dict` specific to a peft strategy